### PR TITLE
Santa is coming to town

### PR DESCRIPTION
**Pull request by danybony**
This addresses https://github.com/novoda/snowy-village-wallpaper/issues/1

During the Christmas day, Santa will visit the town.
He moves in both directions and given that he will be very busy that day, he will disappear for a while (between 5 and 10 seconds) after every round in town.

santa

### CODE REVIEW

**Comment by ryanbateman** [comment]
animade_full1

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/RoundDelay.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.novoda.snowyvillagewallpaper.santa;
+
+import java.util.Random;
+
+class RoundDelay {
+
+    private static final int SECONDS = 1000;
+    private static final int MIN_INTERVAL_BETWEEN_ROUNDS = 5 * SECONDS;
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/RoundDelay.java]: Use `TimeUnit` here instead!

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package com.novoda.snowyvillagewallpaper.santa;
+
+import java.util.Random;
+
+public class SantaTracker {
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java]: :100: separation of concerns FTW

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    private void resetPosition() {
+        y = resetYToCentralHeight();
+        if (direction == Direction.TO_RIGHT) {
+            x = -santaWidth;
+        } else {
+            x = skyWidth;
+        }
+    }
+
+    private float resetYToCentralHeight() {
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java]: Doesn't look central... maybe "initial" is clearer

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        Random random = new Random();
+        return random.nextFloat() * (skyHeight - santaHeight) * SKY_HEIGHT_FRACTION + (skyHeight - santaHeight) * TOP_SKY_FRACTION;
+    }
+
+    public void updatePosition() {
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java]: I'd call it something like `advanceToNextFrame()` or `tick()`, to give it some clearer tie to the time/frames

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        if (direction == Direction.TO_RIGHT) {
+            x += SPEED;
+        } else {
+            x -= SPEED;
+        }
+        if (x > skyWidth || x < -santaWidth) {
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java]: Extract check?

File: app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+    public boolean movingToRight() {
+        return direction == Direction.TO_RIGHT;
+    }
+
+    public boolean isSantaInTown() {
+        return santaSchedule.isSantaInTown();
+    }
+
+    private enum Direction {
```
Comment (rock3r) [inline app/src/main/java/com/novoda/snowyvillagewallpaper/santa/SantaTracker.java]: #enumsmatter ehehe

File: app/src/test/java/com/novoda/snowyvillagewallpaper/santa/SantaScheduleTest.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class SantaScheduleTest {
```
Comment (rock3r) [inline app/src/test/java/com/novoda/snowyvillagewallpaper/santa/SantaScheduleTest.java]: :100: for tests

**Comment by rock3r** [comment]
LGTM :+1: let's :shipit:

**Pull request merged**